open_feature/open_feature_client.py

```diff
@@ -149,34 +149,54 @@ def evaluate_flag_details(
         flag_type: FlagType,
         key: str,
         default_value: typing.Any,
-        evaluation_context: typing.Any = None,
+        evaluation_context: EvaluationContext = None,
         flag_evaluation_options: typing.Any = None,
     ) -> FlagEvaluationDetails:
+        if evaluation_context is None:
+            evaluation_context = EvaluationContext()
         try:
-            args = (
+            # The merge below will eventually pull in Evaluation contexts from Hooks
+            invocation_context = EvaluationContext()
+            invocation_context.merge(ctx1=EvaluationContext(), ctx2=evaluation_context)
+            return self.create_provider_evaluation(
+                flag_type,
                 key,
                 default_value,
-                evaluation_context,
+                invocation_context,
                 flag_evaluation_options,
             )
-            if flag_type is FlagType.BOOLEAN:
-                return self.provider.get_boolean_details(*args)
-
-            elif flag_type is FlagType.NUMBER:
-                return self.provider.get_number_details(*args)
-
-            elif flag_type is FlagType.OBJECT:
-                return self.provider.get_object_details(*args)
-            # Fallback case is a string object
-            elif flag_type is FlagType.STRING:
-                return self.provider.get_string_details(*args)
-            else:
-                raise GeneralError(error_message="Unknown flag type")
-
-        except OpenFeatureError:
+        except OpenFeatureError as ofe:
```

Comment:
I think we will have to catch all exceptions here by the way? Something that Justin / Todd mentioned in the previous review which I think we missed here. One for a separate PR though.

Reply:
That's fine, we just need to handle the error message though as we can't guarantee there is a message in a generic exception, which we use for the error code.